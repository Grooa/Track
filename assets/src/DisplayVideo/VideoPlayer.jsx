import PropTypes from 'prop-types';

const VideoPlayer = (props) => {
  if (!props.url) {
    return <div
      className="placeholder"
      style={{ backgroundColor: '#212121', height: '20em', width: '100%' }} />;
  }

  return <video
    className="object course-video"
    poster={props.poster}
    controls
    controlsList="nodownload">

    <source src={props.url} type="video/mp4" />
    <p>Your browser do not support the HTML5 videoplayer</p>
  </video>;
};

VideoPlayer.propTypes = {
  url: PropTypes.string,
  poster: PropTypes.string,
};


export default VideoPlayer;
