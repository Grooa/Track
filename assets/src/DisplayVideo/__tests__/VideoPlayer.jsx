import { render } from 'enzyme';

import VideoPlayer from '../VideoPlayer';

describe('<VideoPlayer />', () => {
  it('should render placeholder properly', () => {
    const wrapper = render(<VideoPlayer />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render video-container properly', () => {
    const wrapper = render(<VideoPlayer
      url="/somepath/some-video.mp4"
      poster="somepath/some-poster.jpg"/>);

    expect(wrapper).toMatchSnapshot();
  });
});
