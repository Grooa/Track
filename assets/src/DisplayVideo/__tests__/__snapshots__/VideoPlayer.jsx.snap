// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlayer /> should render placeholder properly 1`] = `
<div
  class="placeholder"
  style="background-color:#212121;height:20em;width:100%"
/>
`;

exports[`<VideoPlayer /> should render video-container properly 1`] = `
<video
  class="object course-video"
  controls=""
  controlslist="nodownload"
  poster="somepath/some-poster.jpg"
>
  <source
    src="/somepath/some-video.mp4"
    type="video/mp4"
  />
  <p>
    Your browser do not support the HTML5 videoplayer
  </p>
</video>
`;
